src/internal/adapter/web/WebPresentationModeManager.ts

```diff
@@ -66,8 +66,11 @@ export class WebPresentationModeManager {
   private prepareForPresentationModeChanges() {
     const elements = this._player.element.children;
     for (const element of Array.from(elements)) {
-      if (element.tagName === 'VIDEO' && element.attributes.getNamedItem('src') !== null) {
-        this._element = element as HTMLVideoElement;
+      if (element.tagName === 'VIDEO') {
+        const videoElement = element as HTMLVideoElement;
+        if ((videoElement.src !== null && videoElement.src !== '') || videoElement.srcObject !== null) {
+          this._element = element as HTMLVideoElement;
```

Comment:
There should only be one? If not there is no way of saying if you need the first or the last...

Reply:
The Web SDK may be using multiple `<video>` elements for various reasons:
* Client-side ads play in a separate `<video>` element.
* Some MPEG-DASH streams with different DRM in different periods may play in two `<video>` elements, to enable seamless playback.

...And this is why the Web SDK should support native PIP itself, rather than letting third-party code mess around with its `<video>` element(s). 😛

Reply:
I added the break...